### PR TITLE
Documented new flags for CLI file content specification

### PR DESCRIPTION
**Pull request by abi87**
Documented the new flags which allows to specify file content via CLI. This should finally enable to pass all avalanche configs via flags

### CODE REVIEW

File: docs/build/references/command-line-interface.md [hunk 1]
```diff
@@ -20,6 +20,14 @@ Example JSON config file:
 }
 ```
 
+`--config-file-content` (string):
+
+As an alternative to `--config-file`, it allows specifying base64 encoded config content. Must be used in conjunction with `--config-file-content-type`.
+
+`--config-file-content-type` (string):
+
+Specifies the format of the base64 encoded config content. Required if `--config-file-content` is set.
```
Comment (jpop32) [inline docs/build/references/command-line-interface.md:29]: Might be useful to list which types are allowed. Might just be my ignorance, but without looking into the source it's not entirely clear what should those files types be. Is this like JSON/YAML and such?

File: docs/build/references/command-line-interface.md [hunk 2]
```diff
@@ -126,15 +138,23 @@ Each node runs an HTTP server that provides the APIs for interacting with the no
 
 `--http-tls-cert-file` (string, file path):
 
-This argument specifies the location of the TLS certificate used by the node for the HTTPS server. This must be specified when `--http-tls-enabled=true`. There is no default value.
+This argument specifies the location of the TLS certificate used by the node for the HTTPS server. This must be specified when `--http-tls-enabled=true`. There is no default value. This flag is ignored if `--http-tls-cert-file-content` is specified.
+
+`--http-tls-cert-file-content` (string):
+
+As an alternative to `--http-tls-cert-file`, it allows specifying base64 encoded content of the TLS certificate used by the node for the HTTPS server. This must be specified when `--http-tls-enabled=true`.
```
Comment (jpop32) [inline docs/build/references/command-line-interface.md:145]: Does this take into account that TLS cert and key file is already base64 encoded? Are the users supposed to base64 encode the contents again (with the leading and trailing header), or just use the base64 contents of the file? Might be good to be specific.

Comment (jpop32) [reply docs/build/references/command-line-interface.md:145]: Same question applies to other TLS flags as well.

File: docs/build/references/command-line-interface.md [hunk 3]
```diff
@@ -228,11 +248,19 @@ It means that this node will sample all nodes, not just validators.
 
 `--staking-tls-cert-file` (string, file path):
 
-Avalanche uses two-way authenticated TLS connections to securely connect nodes. This argument specifies the location of the TLS certificate used by the node. By default, the node expects the TLS certificate to be at `$HOME/.avalanchego/staking/staker.crt`.
+Avalanche uses two-way authenticated TLS connections to securely connect nodes. This argument specifies the location of the TLS certificate used by the node. By default, the node expects the TLS certificate to be at `$HOME/.avalanchego/staking/staker.crt`. This flag is ignored if `--staking-tls-cert-file-content` is specified.
+
+`--staking-tls-cert-file-content` (string):
+
+As an alternative to `--staking-tls-cert-file`, it allows specifying base64 encoded of the TLS certificate used by the node.
```
Comment (jpop32) [inline docs/build/references/command-line-interface.md:255]: A possible missing word here `...specifying base64 encoded contents?/file? of the TLS...`

File: docs/build/references/command-line-interface.md [hunk 4]
```diff
@@ -228,11 +248,19 @@ It means that this node will sample all nodes, not just validators.
...
 
 `--staking-tls-key-file` (string, file path):
 
-Avalanche uses two-way authenticated TLS connections to securely connect nodes. This argument specifies the location of the TLS private key used by the node. By default, the node expects the TLS private key to be at `$HOME/.avalanchego/staking/staker.key`.
+Avalanche uses two-way authenticated TLS connections to securely connect nodes. This argument specifies the location of the TLS private key used by the node. By default, the node expects the TLS private key to be at `$HOME/.avalanchego/staking/staker.key`. This flag is ignored if `--staking-tls-key-file-content` is specified.
+
+`--staking-tls-key-file-content` (string):
+
+As an alternative to `--staking-tls-key-file`, it allows specifying base64 encoded of the TLS private key used by the node.
```
Comment (jpop32) [inline docs/build/references/command-line-interface.md:263]: Again a missing word.

**Review by jpop32 — COMMENTED** [review]
Some nits.

File: docs/build/references/command-line-interface.md (hunk 2 above)
Comment (abi87) [reply docs/build/references/command-line-interface.md]: Yes the full file content must be base64 encoded and passed as flag argument. Remarked this in the docs, so hopefully it's clearer

File: docs/build/references/command-line-interface.md (hunk 1 above)
Comment (abi87) [reply docs/build/references/command-line-interface.md]: We currently use [viper ](https://github.com/spf13/viper#reading-config-files) to parse configs, so we can handle JSON, TOML, YAML among others.  I included the viper link to show full list of supported types.

File: docs/build/references/command-line-interface.md (hunk 3 above)
Comment (abi87) [reply docs/build/references/command-line-interface.md]: Done

File: docs/build/references/command-line-interface.md (hunk 4 above)
Comment (abi87) [reply docs/build/references/command-line-interface.md]: Done

**Review by jpop32 — APPROVED** [review]
LGTM

File: docs/build/references/command-line-interface.md [hunk 5]
```diff
@@ -112,7 +120,11 @@ Note that when running with `leveldb`, the node can't read data that was persist
 
 `--genesis` (string):
 
-Path to a JSON file containing the genesis data to use. Ignored when running standard networks (Mainnet, Testnet.) If not given, uses default genesis data. For an example of a JSON representation of genesis data, see [here](https://github.com/ava-labs/avalanchego/blob/master/genesis/genesis_local.go#L16).
+Path to a JSON file containing the genesis data to use. Ignored when running standard networks (Mainnet, Testnet), or when `--genesis-content` is specified. If not given, uses default genesis data. For an example of a JSON representation of genesis data, see [here](https://github.com/ava-labs/avalanchego/blob/master/genesis/genesis_local.go#L16).
```
Comment (yulin-dong) [inline docs/build/references/command-line-interface.md:123]:
```suggestion
Path to a JSON file containing the genesis data to use. Ignored when running standard networks (Mainnet, Fuji Testnet), or when `--genesis-content` is specified. If not given, uses default genesis data. For an example of a JSON representation of genesis data, see [here](https://github.com/ava-labs/avalanchego/blob/master/genesis/genesis_local.go#L16).
```

**Review by yulin-dong — APPROVED** [review]

**Review by yulin-dong — APPROVED** [review]

**Pull request merged**